families/track_and_trade/client/src/fish_form.js

```diff
@@ -0,0 +1,236 @@
+                   oninput: _valChange(
+                       (value) => vnode.state.latitude = parseFloat(value)),
+                   value: vnode.state.latitude
+                 }))),
+
+               m('.col-sm',
+                 _formGroup('Weight (kg)', m('input.form-control', {
+                   type: 'number',
+                   step: 'any',
+                   oninput: _valChange(
+                       (value) => vnode.state.weightInKg = parseFloat(value)),
+                   value: vnode.state.weightInKg
+                 })),
+
+                 _formGroup('Longitude', m('input.form-control', {
+                   type: 'number',
+                   step: 'any',
+                   min: -180,
+                   max: 180,
+                   oninput: _valChange(
+                       (value) => vnode.state.longitude = parseFloat(value)),
+                   value: vnode.state.longitude
+                 })))),
+
+             m('.reporters.form-group',
+               m('label', 'Authorize Reporters'),
+
+               vnode.state.reporters.map((reporter, i) => 
+                 m('.row.mb-2',
+                   m('.col-sm-8',
+                     m('input.form-control', {
+                       type: 'text',
+                       placeholder: 'Add Reporter Private Key',
+                       value: reporter.reporterKey,
+                       oninput: _valChange(
+                         (value) => vnode.state.reporters[i].reporterKey = value),
+                      onblur: () => _updateReporters(vnode, i)
+                     })),
+
+
+                   m('.col-sm-4',
+                     m(MultiSelect, {
+                       label: 'Select Fields',
+                       options: authorizableProperties,
+                       selected: reporter.properties,
+                       onchange: (selection) =>
+                         vnode.state.reporters[i].properties = selection
+                     }))))),
+                          
+             m('.row.justify-content-end.align-items-end',  
+               m('col-2',
+                 m('button.btn.btn-primary',
+                   'Create Record')))))
+  }
+}
+
+const _updateReporters = (vnode, reporterIndex) => {
+  let reporterInfo = vnode.state.reporters[reporterIndex]
+  let lastIdx =  vnode.state.reporters.length - 1
+  if (!reporterInfo.reporterKey && reporterIndex != lastIdx) {
+    vnode.state.reporters.splice(reporterIndex, 1)
+  } else if (reporterInfo.reporterKey && reporterIndex == lastIdx) {
+    vnode.state.reporters.push({
+      reporterKey: '',
+      properties: []
+    })
+  }
+}
+
+const _handleSubmit = (signingKey, state) => {
+  let txnFields = {
+    recordId: state.serialNumber,
+    recordType: 'fish',
+    properties: [
+      {
+        name: 'species',
+        value: state.species,
+        type: 1, // STRING
+      },
+      {
+        name: 'length',
+        value: state.lengthInCM,
+        type: 3, // FLOAT
```

Comment:
I had considered using integers for these and making the units something like _mm_ and _g_. The UI would display _cm_ and _kg_ since those are more human friendly, but bon convert the input float to an integer in the background before submitting.

Might be more trouble than it is worth though. Happy to make these fields floats.

Reply:
Keep in mind, these are placeholders for when the record type is actually in place.  Will update when the transaction submission is ready to go.

Reply:
Sounds fine to me.